### PR TITLE
Initialize transactions once

### PR DESCRIPTION
**Pull request by StephenButtolph**
## Why this should be merged

Currently, when creating a block, we re-initialize all transactions being placed into the block. This is inefficient, as the transactions are already initialized. However, this can also introduce racy behavior. In a concurrent setting we generally want to be able to treat transactions as immuable. However calling `.ID` on a transaction that could be re-initialized in a different goroutine would be racy.

## How this works

Ensure that transactions are only initialized once, during creation or parsing.

## How this was tested

- [X] CI

### CODE REVIEW

File: vms/avm/txs/parser.go [hunk 1]
```diff
@@ -157,19 +146,3 @@ func parse(cm codec.Manager, signedBytes []byte) (*Tx, error) {
...
-
-	unsignedBytesLen, err := cm.Size(CodecVersion, &tx.Unsigned)
-	if err != nil {
-		return fmt.Errorf("couldn't calculate UnsignedTx marshal length: %w", err)
-	}
-
-	unsignedBytes := signedBytes[:unsignedBytesLen]
-	tx.SetBytes(unsignedBytes, signedBytes)
-	return nil
-}
```
Comment (StephenButtolph) [inline vms/avm/txs/parser.go:175]: This is 100% duplicated from `tx.Initialize`. I have no recollection as to why this was done... but something seems to have been a mistake imo.

File: vms/avm/txs/parser.go [hunk 2]
```diff
@@ -130,14 +127,6 @@ func (p *parser) ParseGenesisTx(bytes []byte) (*Tx, error) {
 	return parse(p.gcm, bytes)
 }
 
-func (p *parser) InitializeTx(tx *Tx) error {
-	return initializeTx(p.cm, tx)
-}
-
-func (p *parser) InitializeGenesisTx(tx *Tx) error {
-	return initializeTx(p.gcm, tx)
-}
```
Comment (StephenButtolph) [inline vms/avm/txs/parser.go:139]: `Tx` has an `Initialize` function and we already expose the `codec` - so these functions aren't needed. Additionally, `InitializeTx` was only used in tests.

File: vms/avm/block/parser.go [hunk 3]
```diff
@@ -88,21 +84,3 @@ func parse(cm codec.Manager, bytes []byte) (Block, error) {
...
 	return blk, blk.initialize(bytes, cm)
 }
-
-func (p *parser) InitializeBlock(block Block) error {
-	return initialize(block, p.Codec())
-}
-
-func (p *parser) InitializeGenesisBlock(block Block) error {
-	return initialize(block, p.GenesisCodec())
-}
```
Comment (StephenButtolph) [inline vms/avm/block/parser.go:98]: These functions were never used.

File: vms/avm/block/standard_block.go [hunk 4]
```diff
@@ -88,5 +88,16 @@ func NewStandardBlock(
 		Time:         uint64(timestamp.Unix()),
 		Transactions: txs,
 	}
-	return blk, initialize(blk, cm)
```
Comment (StephenButtolph) [inline vms/avm/block/standard_block.go:91]: Calling `initialize` here used to call `blk.initialize` which would re-initialize the transactions. Now we explicitly do not re-initialize the transactions.

**Review by dhrubabasu — APPROVED** [review]

**Review by danlaine — APPROVED** [review]

**Review by abi87 — APPROVED** [review]

**Pull request merged**